src/renderer/components/pages/Channel.tsx

```diff
@@ -28,7 +28,11 @@ export interface ChannelComponentProps {
   channel: PublicChannel
   channelSettingsModal: ReturnType<typeof useModal>
   channelInfoModal: ReturnType<typeof useModal>
-  messages: { [date: string]: DisplayableMessage[][] }
+  messages: {
+    count: number
+    groups: { [date: string]: DisplayableMessage[][] }
```

Comment:
I'm not 100% sure if storing groups in an object enumarted by date will not mess up the relative order of groups.
I know `Map` maintains order of insertions, but I'm really not sure about objects.
Maybe you should use sth like: 
```
[ 
[string, DisplayableMessage[][] ]
]
```
instead ?

Reply:
I've got tests for selectors and they seem fine. Also we've been chatting with Bartek (and even Holmes) a few times and it was working well every time